test/micro/org/openjdk/bench/java/lang/MinMaxLoopBench.java

```diff
@@ -0,0 +1,180 @@
+import java.util.concurrent.TimeUnit;
+import java.util.stream.LongStream;
+
+@Warmup(iterations = 3, time = 5)
+@Measurement(iterations = 4, time = 5)
+@Fork(2)
+@OutputTimeUnit(TimeUnit.MILLISECONDS)
+@BenchmarkMode(Mode.Throughput)
+@State(Scope.Thread)
+public class MinMaxLoopBench
+{
+    @Param({"100", "1000", "10000"})
+    int size;
+
+    /**
+     * Probability of one of the min/max branches being taken.
+     * For max, this value represents the percentage of branches in which
+     * the value will be bigger or equal than the current max.
+     * For min, this value represents the percentage of branches in which
+     * the value will be smaller or equal than the current min.
+     */
+    @Param({"50", "80", "100"})
+    int probability;
+
+    int[] minIntA;
+    int[] minIntB;
+    long[] minLongA;
+    long[] minLongB;
+    int[] maxIntA;
+    int[] maxIntB;
+    long[] maxLongA;
+    long[] maxLongB;
+
+    @Setup
+    public void setup() {
+        final long[][] longs = distributeLongRandomIncrement(size, probability);
+        maxLongA = longs[0];
+        maxLongB = longs[1];
+        maxIntA = toInts(maxLongA);
+        maxIntB = toInts(maxLongB);
+        minLongA = negate(maxLongA);
+        minLongB = negate(maxLongB);
+        minIntA = toInts(minLongA);
+        minIntB = toInts(minLongB);
+    }
+
+    static long[] negate(long[] nums) {
+        return LongStream.of(nums).map(l -> -l).toArray();
+    }
+
+    static int[] toInts(long[] nums) {
+        return Arrays.stream(nums).mapToInt(i -> (int) i).toArray();
+    }
+
+    static long[][] distributeLongRandomIncrement(int size, int probability) {
+        long[][] result;
+        int aboveCount, abovePercent;
+
+        // Iterate until you find a set that matches the requirement probability
+        do {
+            long max = ThreadLocalRandom.current().nextLong(10);
+            result = new long[2][size];
+            result[0][0] = max;
+            result[1][0] = max - 1;
+
+            aboveCount = 0;
+            for (int i = 1; i < result[0].length; i++) {
+                long value;
+                if (ThreadLocalRandom.current().nextLong(101) <= probability) {
+                    long increment = ThreadLocalRandom.current().nextLong(10);
+                    value = max + increment;
+                    aboveCount++;
+                } else {
+                    // Decrement by at least 1
+                    long decrement = ThreadLocalRandom.current().nextLong(10) + 1;
+                    value = max - decrement;
+                }
+                result[0][i] = value;
+                result[1][i] = max;
+                max = Math.max(max, value);
+            }
+
+            abovePercent = ((aboveCount + 1) * 100) / size;
+        } while (abovePercent != probability);
+
+        return result;
+    }
+
+    @Benchmark
+    public int[] intLoopMin() {
+        final int[] result = new int[size];
```

Comment:
it would be better to not have this allocation here unless is what you want to measure 
i.e. allocation and assignment in a fresh new array which would escape